### PR TITLE
config: validate config.json file

### PR DESCRIPTION
**Pull request by sreeram-boyapati**
Minio Client validates only config host add parameters
    Not the config file. Use the same utils to validate config file|
#1543

### CODE REVIEW

**Comment by harshavardhana** [comment]
Can you also add how the errors would look like here in the commit logs?

File: config-validate.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"bytes"
+	"container/list"
```
Comment (harshavardhana) [inline config-validate.go]: It is better to use a simple slice of error messages here, container/list is an overkill, even the order and popping can be done easily. 

https://github.com/golang/go/wiki/SliceTricks

**Comment by sreeram-boyapati** [comment]
I will add how the error messages will look like and slices instead of container/list. But they do not stop the execution if validation fails. Is it ok?

**Comment by harshavardhana** [comment]
> I will add how the error messages will look like and slices instead of container/list. But they do not stop the execution if validation fails. Is it ok?

They should stop, we can error out properly.

**Pull request merged**